### DISCUSSION TITLE
Add attributes to the correct session inside on successful authentication

### DISCUSSION THREAD

**Discussion by Jumborner** [symfony/symfony#43043] [Q&A] [answered]
Hello, using vue front end with symfony back end. I need to add some attributes to the user session if the authentication was successful. I have a custom authenticator for the user authentication. After the login, the main menu will load from the back end, which is taken from the database using the session's attribute.

Right now, upon successful login the menu will not load, since the attribute which is needed is missing from the session in symfony.

```
 public function onAuthenticationSuccess(Request $request, TokenInterface $token, string $firewallName): ?Response
    {
        //$_SESSION['resourceId'] = $this->resourceId;
        $request->getSession()->set('resourceId', $this->resourceId);
        return new JsonResponse("Login success!"); } 
```

It seems that the session changes after authentication, and my attempts to add the attributes come in too early.
Any suggestions are welcome

**Comment by jfsenechal** [comment]
Hi
is $this->resourceId empty ?

**Reply by Jumborner** [reply]
> Hi
> is $this->resourceId empty ?

No it is not. Even if I hardcore a string, once I try to get it after the authentication it says that the attribute is undefined.

**Comment by jfsenechal** [comment, marked as answer]
it's weird i tried and it works 

```php
 public function onAuthenticationSuccess(Request $request, TokenInterface $token, string $firewallName): ?Response
    {
        $request->getSession()->set('zeze','zozo');
       
```

In controller

```php   
use Symfony\Component\HttpFoundation\Session\SessionInterface;

 public function __construct(
        SessionInterface $session
    ) {
        $this->session = $session;
    }

 /**
     * @Route("/", name="app_index", methods={"GET"})
     */
    public function index(): Response
    {
        dump($this->session->get('zeze'));
      
    }
```

**Reply by Jumborner** [reply]
Using the session session interface worked. Previously i have tried to get the session using RequestStack or $_SESSION() which returned me undefined.
